### DISCUSSION TITLE
Could you add a devcontainer configuration to this repo?

### DISCUSSION THREAD

**Discussion by spenserblack** [github-linguist/linguist#6473] [Q&A] [answered]
None of the issue types seem applicable, so this seems like the best place to ask.

I like to use GitHub Codespaces, especially for large repos that might take up space locally, or require a lot of extra installed tools.

The base codespace template is good enough to get started, but a custom `.devcontainer.json` or `.devcontainer/devcontainer.json` could

- use the ruby image to get the desired ruby version installed
- use the "docker-in-docker" feature to pre-install docker
- run `scripts/bootstrap`
- open `CONTRIBUTING.md` by default to hopefully encourage reading it :wink:

So, yeah, it's not *necessary* since the default config already comes with Ruby, but it helps skip past the "Dependencies" and "Getting Started" parts of the contributing doc and go straight to the actual contribution instructions.

Edit: looks like there was a bit of work done [here](https://github.com/github-linguist/linguist/pull/6205), but it doesn't seem like the PR author is going to continue soon.

**Comment by lildude** [comment, marked as answer]
Definitely! I started this a while back when Codespaces was first launched before pre-builds were available as it would give everyone a consistent dev env but found it took waaaaay too long versus using the default container which is quick and does everything out the box so I've not looked at it since (eagle eyed viewers may have even spotted the branch hanging around for a while 😁). Now we've got pre-builds we should be able to speed this up considerably.

I'll try carve out some time in the next few months (day job takes priority, and it's not linguist). If you've got the time and resources, feel free to make a start on it, and we can work on it together.

**Reply by spenserblack** [reply]
Awesome, thanks!

**Reply by spenserblack** [reply]
:thinking: It might be worth having multiple configurations. I think the majority of contributions don't need docker installed, for example. Could be like `slim`, `full`, `fuller` or something. But I'm mainly just thinking this way because I don't use prebuilds. I guess it would be equally fast setup regardless of features if the container is prebuilt?

**Reply by lildude** [reply]
Yup, pre-builds should remove the need for different and potentially confusing configurations as most of the work should already have been done.
